urllib3/sessionmanager.py

```diff
@@ -0,0 +1,78 @@
+from .util.retry import Retry, handle_redirect
+from .request import RequestMethods, Request
+from .poolmanager import PoolManager, ProxyManager
+from .util.context import Context
+
+
+class SessionManager(RequestMethods):
+    """
+    Allows arbitrary requests while maintaining browser-session-like
+    context across those requests.
+
+    :param headers:
+        Headers to include with all requests, unless other headers are
+        given explcitly
+
+    :param cookie_jar:
+        A predefined :class:`CookieJar` object to use in the session;
+        if not provided, a new one will be created.
+
+    :param \**pm_kwargs:
+        Additional paramaters to be passed to the encapsulated
+        :class:`.PoolManager` or :class:`.ProxyManager` instance.
+    """
+    def __init__(self, headers=None, cookie_jar=None, **pm_kwargs):
+
+        super(SessionManager, self).__init__(headers=headers)
+
+        # Detect if we should create a PoolManager or a ProxyManager
+        if pm_kwargs.get('proxy_url', None) is None:
```

Comment:
There are a few places where we distinguish between `None` and `False` as specific behaviour. If we're just checking presence, a `if 'proxy_url` in pm_kwargs:` should be sufficient.

That said, I would prefer to avoid this flow altogether and instead pass in a pre-made Manager to the constructor. We can add classmethod helpers as shortcuts for creating SessionManager objects with uninitialized Manager components later.